client/src/pages/Tasks.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import { useNetworkStatus } from 'react-offix-hooks';
+import { add } from 'ionicons/icons';
+import { 
+  IonHeader, 
+  IonPage, 
+  IonTitle, 
+  IonToolbar, 
+  IonSegment, 
+  IonSegmentButton, 
+  IonLabel, 
+  IonButtons, 
+  IonIcon, 
+  IonButton, 
+  IonMenuButton, 
+  IonFooter, 
+  IonBadge,
+  IonLoading 
+} from '@ionic/react';
+import TaskList from '../components/TaskList';
+import { GET_TASKS } from '../gql/queries';
+import { subscriptionOptions,  } from '../helpers';
+import { useSubscribeToMore, useOfflineQueueListener } from '../hooks';
+import { ITask } from '../declarations';
+import { Empty } from '../components/Empty';
+
+const Task: React.FC = () => {
+
+  const isOnline = useNetworkStatus();
+  const offlineQueue = useOfflineQueueListener();
+  const [tasks, setTasks] = useState<[ITask]>(null!);
+  const { loading, error, data, subscribeToMore } = useQuery(GET_TASKS);
+  useSubscribeToMore({ options: subscriptionOptions, subscribeToMore});
+
+  useEffect(() => {
+    if (data) {
+      setTasks(data.allTasks);
```

Comment:
Out of interest. Does this cause additional renders that could be avoided?

Surely, the `useQuery` hook will cause a rerender when `data` comes back, then we do our own thing here that calls `setTasks`. Does this cause another render I wonder?

Couldn't you avoid the `useState` and `useEffect` calls with something like this?

```ts
const { loading, error, data, subscribeToMore } = useQuery(GET_TASKS);
const tasks = data && data.allTasks
```

I'm honestly not strong on this but I'm just trying to find way to keep this stuff as simple as possible

Reply:
Completely overcooked that. That's definitely a far cleaner and far simpler way of doing things.